### DISCUSSION TITLE
Unable to locate the METplus sample test data directory

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2118] [Training] [answered]
I am setting up for participating in the METPLUS ADVANCED TRAINING SERIES. I'm setting up the tutorial environment (bash) on the DoD HPC. When editing the METplus v5.0.0 Tutorial Setup Script I think I have the correct paths for the following:
METPLUS_BUILD_BASE=/p/home/jraby/MET/METplus2
MET_BUILD_BASE=/p/home/jraby/MET/METplus2

The above two paths point to the location of the METplus V5.0.1 singularity container (metplus.simg). When the container is running it should execute point_stat to display the usage and also execute the command "which run_metplus.py", but I have not tested this pending my editing the script to set the correct path for the METPLUS_DATA variable. Where is the sample test data directory? I assume that it is located inside the container, but I haven't been able to find it. Thanks.

**Comment by CPKalb** [comment]
Hi John:

Apologies for the slow response.  It took a bit of time to figure this out on my end.  METplus sample data for the container can be accessed [here](https://hub.docker.com/r/dtcenter/metplus-data).  Are you able to access to this?

Christina

**Reply by jwraby** [reply]
Hi Christina - I attempted to pull the image file as I do the MET and METplus containers on the HPC, but it has an error. See the following dialog:
/p/home/jraby/MET/METplus_Data
(base) [~/MET/METplus_Data]$ singularity pull docker://dtcenter/metplus-data
FATAL:   While making image from oci registry: failed to get checksum for docker://dtcenter/metplus-data: Error reading manifest latest in docker.io/dtcenter/metplus-data: manifest unknown: manifest unknown.
I am not sure how to resolve this error.

**Reply by JohnHalleyGotway** [reply]
@jwraby, I was able to replicate the error you saw:
```
singularity pull docker://dtcenter/metplus-data
FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://dtcenter/metplus-data: reading manifest latest in docker.io/dtcenter/metplus-data: manifest unknown: manifest unknown
```
However when I specify a "tag" name, the command does actually work. For example:
```
singularity pull docker://dtcenter/metplus-data:5.0-marine_and_cryosphere
```
And you can see the list of tags [here](https://hub.docker.com/r/dtcenter/metplus-data/tags).

The test data for ALL the use cases is rather large. As such, its split across several [tar files](https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.0/) and there's a corresponding [docker image tag](https://hub.docker.com/r/dtcenter/metplus-data/tags) for each.

I'll note that I have not personally run these commands through Singularity myself. But just wanted to point out that detail.

**Reply by jwraby** [reply, marked as answer]
Thanks for the tip on modifying the pull command by adding the "tag" name. This worked for me. The singularity container was installed on the HPC. I found that it will not "execute" or "run", but I was able to use singularity shell to look at the data directories and found the data at:
/data/input/METplus_Data/model_applications/marine_and_cryosphere
I assume that this is the data needed for the METplus sample test data. 
Thanks to Christina too for debugging the issue and locating the sample data.
